orchagent/aclorch.cpp

```diff
@@ -157,18 +166,320 @@ static acl_ip_type_lookup_t aclIpTypeLookup =
+{
+    return m_aclAcitons;
+}
+
+AclTableTypeBuilder& AclTableTypeBuilder::withName(string name)
+{
+    m_tableType.m_name = name;
+    return *this;
+}
+
+AclTableTypeBuilder& AclTableTypeBuilder::withBindPointType(sai_acl_bind_point_type_t bpointType)
+{
+    m_tableType.m_bpointTypes.insert(bpointType);
+    return *this;
+}
+
+AclTableTypeBuilder& AclTableTypeBuilder::withMatch(shared_ptr<AclTableMatchInterface> match)
+{
+    m_tableType.m_matches.emplace(match->getId(), match);
+    return *this;
+}
+
+AclTableTypeBuilder& AclTableTypeBuilder::withAction(sai_acl_action_type_t action)
+{
+    m_tableType.m_aclAcitons.insert(action);
+    return *this;
+}
+
+AclTableType AclTableTypeBuilder::build()
+{
+    auto tableType = m_tableType;
+    m_tableType = AclTableType();
+    return tableType;
+}
+
+bool AclTableTypeParser::parse(const std::string& key,
+                               const vector<swss::FieldValueTuple>& fieldValues,
+                               AclTableTypeBuilder& builder)
+{
+    builder.withName(key);
+
+    for (const auto& fieldValue: fieldValues)
+    {
+        auto field = to_upper(fvField(fieldValue));
+        auto value = to_upper(fvValue(fieldValue));
+
+        SWSS_LOG_DEBUG("field %s, value %s", field.c_str(), value.c_str());
+
+        if (field == ACL_TABLE_TYPE_MATCHES)
+        {
+            if (!parseAclTableTypeMatches(value, builder))
+            {
+                return false;
+            }
+        }
+        else if (field == ACL_TABLE_TYPE_ACTIONS)
+        {
+            if (!parseAclTableTypeActions(value, builder))
+            {
+                return false;
+            }
+        }
+        else if (field == ACL_TABLE_TYPE_BPOINT_TYPES)
+        {
+            if (!parseAclTableTypeBindPointTypes(value, builder))
+            {
+                return false;
+            }
+        }
+        else
+        {
+            SWSS_LOG_ERROR("Unknown field %s: value %s", field.c_str(), value.c_str());
+            return false;
+        }
+    }
+
+    return true;
+}
+
+bool AclTableTypeParser::parseAclTableTypeMatches(const std::string& value, AclTableTypeBuilder& builder)
+{
+    auto matches = tokenize(value, comma);
+    set<sai_acl_range_type_t> saiRangeTypes;
+
+    for (const auto& match: matches)
+    {
+        auto matchIt = aclMatchLookup.find(match);
+        auto rangeMatchIt = aclRangeTypeLookup.find(match);
+
+        if (rangeMatchIt != aclRangeTypeLookup.end())
+        {
+            auto rangeType = rangeMatchIt->second;
+            saiRangeTypes.insert(rangeType);
+        }
+        else if (matchIt != aclMatchLookup.end())
+        {
+            auto saiMatch = AclEntryFieldToAclTableField(matchIt->second);
+            builder.withMatch(make_unique<AclTableMatch>(saiMatch));
+        }
+        else
+        {
+            SWSS_LOG_ERROR("Unknown match %s", match.c_str());
+            return false;
+        }
+    }
+
+    if (!saiRangeTypes.empty())
```

Comment:
Why didn't put this ;ogic in Line 392?

Reply:
On Line 392 we fill saiRangeTypes. Once we finished the loop on 384 I check whether it is empty or not.
I could actually remove this "if" and set empty range always, however I prefered to not even pass an empty SAI_ACL_ENTRY_ATTR_FIELD_ACL_RANGE_TYPE to SAI.